singleflight/singleflight.go

```diff
@@ -204,6 +204,9 @@ func (g *Group) doCall(c *call, key string, fn func() (interface{}, error)) {
 // an earlier call to complete.
 func (g *Group) Forget(key string) {
 	g.mu.Lock()
+	if g.m == nil {
+		g.m = make(map[string]*call)
+	}
```

Comment:
Alternatively we could also do:

```suggestion
	if g.m == nil {
		g.mu.Unlock()
		return
	}
```

I went for the lazy map init for uniformity.